### PR TITLE
chore(core): do not delete snapshot folder

### PR DESCRIPTION
**Pull request by ideoma**
For better integration when `SNAPSHOT COMPLETE` command is executed do not delete `snapshot` folder but only clear the internal `db` folder to delete the snapshot

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 70 / 76 (92.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/DatabaseSnapshotAgent.java|70|76|92.11%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 70 / 76 (92.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/DatabaseSnapshotAgent.java|70|76|92.11%|

File: core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java [hunk 1]
```diff
@@ -325,4 +211,126 @@ public static void recoverSnapshot(CairoEngine engine) {
...
+                        // Copy metadata files for all tables.
+
+                        while (cursor.hasNext()) {
+                            CharSequence tableName = record.getStr(tableNameIndex);
+                            if (
+                                    TableUtils.isValidTableName(tableName, tableName.length())
+                                            && ff.exists(path.of(configuration.getRoot()).concat(tableName).concat(TableUtils.META_FILE_NAME).$())
+                            ) {
+                                path.of(configuration.getSnapshotRoot()).concat(configuration.getDbDirectory());
+                                LOG.info().$("including in the snapshot [table=").$(tableName).$(']').$();
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java:281]: this is unfortunate English. "including in the snapshot" doesn't actually specify the snapshot.

```suggestion
                                LOG.info().$("preparing for snapshot [table=").$(tableName).I$();
```

File: core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java [hunk 2]
```diff
@@ -325,4 +211,126 @@ public static void recoverSnapshot(CairoEngine engine) {
...
+
+                                TableReader reader = engine.getReaderForStatement(executionContext, tableName, "snapshot");
+                                snapshotReaders.add(reader);
+
+                                path.trimTo(snapshotLen).concat(tableName).slash$();
+                                if (ff.mkdirs(path, configuration.getMkDirMode()) != 0) {
+                                    throw CairoException.instance(ff.errno()).put("Could not create [dir=").put(path).put(']');
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java:288]: `errno` missing

File: core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java [hunk 3]
```diff
@@ -325,4 +211,126 @@ public static void recoverSnapshot(CairoEngine engine) {
...
+                                mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                                reader.getTxFile().dumpTo(mem);
+                                mem.close(false);
+                                // Copy _cv file.
+                                path.trimTo(rootLen).concat(TableUtils.COLUMN_VERSION_FILE_NAME).$();
+                                mem.smallFile(ff, path, MemoryTag.MMAP_DEFAULT);
+                                reader.getColumnVersionReader().dumpTo(mem);
+                                mem.close(false);
+                            } else {
+                                LOG.error().$("cannot include table into snapshot, invalid table name or missing metadata [table=").$(tableName).I$();
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java:308]:
```suggestion
                                LOG.error().$("skipping, invalid table name or missing metadata [table=").$(tableName).I$();
```

File: core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java [hunk 4]
```diff
@@ -325,4 +211,126 @@ public static void recoverSnapshot(CairoEngine engine) {
...
+                        // Flush dirty pages and filesystem metadata to disk
+                        if (ff.sync() != 0) {
+                            throw CairoException.instance(ff.errno()).put("Could not sync");
+                        }
+
+                        LOG.info().$("snapshot copying finished").$();
+                    } catch (Throwable e) {
+                        unsafeReleaseReaders();
+                        LOG.error()
+                                .$("snapshot prepare error [e=").$(e)
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java:326]:
```suggestion
                                .$("snapshot error [e=").$(e)
```

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java (hunk 2 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java:288]: I don't understand, errno is present here `CairoException.instance(ff.errno())`

Comment (bluestreak01) [reply core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java:288]: sorry! did not notice!

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 70 / 76 (92.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/DatabaseSnapshotAgent.java|70|76|92.11%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 70 / 76 (92.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/DatabaseSnapshotAgent.java|70|76|92.11%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 70 / 76 (92.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/DatabaseSnapshotAgent.java|70|76|92.11%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**